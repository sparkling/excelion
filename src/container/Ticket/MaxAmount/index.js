import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledInput = styled.input `
  width: 7em;
  font-size: 1em;
  padding: 0.5em;
  text-align: center
`

const MaxAmount = ({ value }) =>
  (<StyledInput
      value={value}
      type="text"
      name="maxAmount"
      readOnly="disabled"
    />);


MaxAmount.propTypes = {
  value: PropTypes.string
}

export default MaxAmount;
