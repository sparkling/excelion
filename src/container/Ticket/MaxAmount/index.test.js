import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import MaxAmount from './index.js';

const renderComponent = (props = {}) => shallow(<MaxAmount {...props} />);
const defaultRender = renderComponent({ onChange:()=>{} }).dive();

describe('<MaxAmount />', () => {

  it('should render the same snapshot', () => {
    const tree = renderer.create(<MaxAmount onChange={ ()=>{} } />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an input field', () => {
    expect(defaultRender.find('input').length).toBe(1);
  });

  it('should render the value', () => {
    const value="Â£50M EUR";
    expect(renderComponent({onChange:()=>{}, value:value}).dive().prop('value')).toBe(value);
  });

  it('should render readOnly="disabled"', () => {
    expect(defaultRender.prop('readOnly')).toBe('disabled');
  });

});
