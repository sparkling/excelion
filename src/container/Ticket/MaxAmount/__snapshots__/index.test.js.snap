// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaxAmount /> should render the same snapshot 1`] = `
.c0 {
  width: 7em;
  font-size: 1em;
  padding: 0.5em;
  text-align: center;
}

<input
  className="c0"
  name="maxAmount"
  readOnly="disabled"
  type="text"
  value={undefined}
/>
`;
